Public/Src/Engine/Cache/Artifacts/LocalDiskContentStore.cs

```diff
@@ -400,7 +399,8 @@ private Possible<Unit> CreateSymlinkIfNotExistsOrTargetMismatch(AbsolutePath sym
                         }
 
                         return openResult.Succeeded;
-                    });
+                    },
+                    numberOfAttempts: OperatingSystemHelper.IsUnixOS ? 1 : 3);

```

Comment:
I don't know what's the rational for retrying this operation.  

For example, here's a possible (and realistic) call stack up to this point:
```
at BuildXL.Engine.Cache.Artifacts.LocalDiskContentStore.TryCreateHandleToFile(String expandedPath, Boolean forUsnQuery, Boolean createHandleWithSequentialScan) in \.\Public\Src\Engine\Cache\Artifacts\LocalDiskContentStore.cs:line 393
   at BuildXL.Engine.Cache.Artifacts.LocalDiskContentStore.TryDiscoverAsync(FileArtifact fileArtifact, ExpandedAbsolutePath path, Boolean ignoreKnownContentHash, Boolean createHandleWithSequentialScan) in \.\Public\Src\Engine\Cache\Artifacts\LocalDiskContentStore.cs:line 467
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BuildXL.Engine.Cache.Artifacts.LocalDiskContentStore.TryDiscoverAsync(FileArtifact fileArtifact, ExpandedAbsolutePath path, Boolean ignoreKnownContentHash, Boolean createHandleWithSequentialScan)
   at BuildXL.Scheduler.Artifacts.FileContentManager.TryLoadAvailableContentAsync(OperationContext operationContext, PipInfo pipInfo, Boolean materializingOutputs, IReadOnlyList`1 filesAndContentHashes, Action`1 onFailure, Action`2 onContentUnavailable, Boolean onlyLogUnavailableContent, PipArtifactsState state) in \.\Public\Src\Engine\Scheduler\Artifacts\FileContentManager.cs:line 2291
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BuildXL.Scheduler.Artifacts.FileContentManager.TryLoadAvailableContentAsync(OperationContext operationContext, PipInfo pipInfo, Boolean materializingOutputs, IReadOnlyList`1 filesAndContentHashes, Action`1 onFailure, Action`2 onContentUnavailable, Boolean onlyLogUnavailableContent, PipArtifactsState state)
   at BuildXL.Scheduler.Artifacts.FileContentManager.TryLoadAvailableContentAsync(OperationContext operationContext, PipInfo pipInfo, PipArtifactsState state) in \.\Public\Src\Engine\Scheduler\Artifacts\FileContentManager.cs:line 2173
   at BuildXL.Scheduler.Artifacts.FileContentManager.PlaceFilesAsync(OperationContext operationContext, PipInfo pipInfo, PipArtifactsState state, Boolean materialize) in \.\Public\Src\Engine\Scheduler\Artifacts\FileContentManager.cs:line 1958
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at BuildXL.Scheduler.Artifacts.FileContentManager.PlaceFilesAsync(OperationContext operationContext, PipInfo pipInfo, PipArtifactsState state, Boolean materialize)
   at BuildXL.Scheduler.Artifacts.FileContentManager.TryMaterializeArtifactsCore(PipInfo pipInfo, OperationContext operationContext, PipArtifactsState state, Boolean materializatingOutputs, Boolean isDeclaredProducer) in \.\Public\Src\Engine\Scheduler\Artifacts\FileContentManager.cs:line 1548
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
``` 
If you look at `FileContentManager.TryLoadAvailableContentAsync`, you can see that it's perfectly fine if `LocalDiskContentStore.TryDiscoverAsync` doesn't succeed (e.g., because the file in question doesn't exist), in which case retrying to open file handle here become a significant waste of time.